qualtran/surface_code/quantum_error_correction_scheme.py

```diff
@@ -0,0 +1,90 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import attr
+import numpy as np
+from attrs import frozen
+
+from qualtran.surface_code.physical_parameters import PhysicalParameters
+
+
+@frozen
+class QuantumErrorCorrectionScheme:
+    """QuantumErrorCorrectionScheme represents a quantum error correction scheme.
+
+        QuantumErrorCorrectionScheme provides estimates for the logical error rate,
+        number of physical qubits and the logical time step given a code distance and
+        physical assumptions.
+
+    Attributes:
+        error_rate_scaler: Logical error rate coefficient.
+        error_rate_threshold: Logical error rate threshold.
```

Comment:
Please document these constants in terms of the mathematical formula in which they appear. I don't think "error_rate_scaler" is a common term and "error_rate_threshold" would imply that this is a property of the code whereas in reality this is yet another fuzzy parameter that captures the empirical behavior of the error suppression. 

Note that we measure/target/think about the parameter $\Lambda$ which is $p*/p$ instead of factoring things into the "physical error rate" and the "error rate threshold". Lambda is an experimentally measurable and relevant property.

Reply:
is $\Lambda$ equal to $\frac{p*}{p}$ or to $\frac{1}{p^*}$?

Reply:
The definition of lambda definitely includes the physical error rate $p$, so it can't just be $\log p^*$.

you can use https://arxiv.org/abs/2207.06431 as a reference for lambda. It's the error at d divided by the error at d+2; i.e. how much suppression you get by increasing the code distance.